### DISCUSSION TITLE
Prompty copying not working anymore?

### DISCUSSION THREAD

**Discussion by Tillerz** [vladmandic/sdnext#1355] [Q&A]
Hi,

is this only me?

I am on version: 276fb6a0 Wed Jun 7 17:45:39 2023 -0400

When I load an image into "process image" and then click "--> Text", it copies all the settings, but it deletes the prompt and negative prompt, not filling in the data from the image.

EDIT1: Also I just see: after copying it shows the Upscaler as `Latent"]` ... looks like there's something broken with the JSON parser now?

EDIT2: And when I try to change the Upscaler, the list opens, I can click an upscale but it doesn't select it and close the list again. Looks like this JavaScript is broken, too. After that the UI is wrecked (not responding to anything) and I have to do a <ctrl-f5> refresh in the browser.

**Comment by ghost** [comment]
I had problems with the "load last generation params" button. Sometimes it just does not work and I dont know why, maybe this is related? It was really a nice function, if I had to reload the ui I had myself covered with this button but not anymore.

Something was changed about the upscalers I guess because some people have submit issues about custom upscalers not working anymore, that might be it, too

**Reply by Tillerz** [reply]
Yes, that button is broken, too.

**Comment by vladmandic** [comment]
i cannot reproduce issues with restore params button - if this is reproducible at will on your system, describe exact steps so i can try?

regarding upscalers, that's different - there was as change in a1111 that i've ported and that breaks custom upscalers, i need to take a closer look at that.

**Reply by Tillerz** [reply]
Start server, open SD, go to process image tab, select image via file dialog, --> Text

Boom, prompts empty and Upscaler name wrong, ending with "]

**Reply by Tillerz** [reply]
Did you try with an older image? Maybe the prompt embedding into the images changed and that's why it's broken now (for older images)? Just a wild guess.

**Reply by Tillerz** [reply]
And for the **restore params** button (the left one that looks like a book): just the same. Start server, open SD, Click restore button. everything gets filled in except prompt and negative prompt.

Loading saved styles still works fine.

**Comment by Tillerz** [comment]
I just checked the embedded data in the image:

Yesterday:

`(UNICODE  ( 1 w o m a n ,   s o l o ) ,   m e d i e v a l ...`

Today:

`(UNICODE  { " p r o m p t " :   " ( 1 w o m a n ,   s o l o ) ,   m e d i e v al ...`

That's why?

But I just tried to load info from an old image and a new image: doesn't work for any. Prompts stay empty.

On the Process Image page it also says:

```
Prompt: {"prompt": "(1woman, solo), medieval, fantasy, stunning intricate full color photo, (beautiful 40yo mature __origin__ adult fae woman __name__ with (__face__:1.2) and (__wildcards/tt-hair__:1.2) ), [:(highly detail face: 1.2):0.1], (__wildcards/eyeliner__),
```

Shouldn't it NOT have the json stuff in there, JUST the prompt starting with `(1woman...` ?

**Comment by vladmandic** [comment]
i was working on that code yesterday. and yes, internally there are jsons where previously there were none. but exif in all of my tests is correct. i just cant reproduce this using either 2 month old image or image i just generated right now. and i've tried in txt2img and extras.

**Reply by Tillerz** [reply]
Ok, I am going to set up a1111 fresh once more (which I already did this morning after the error appeared). But I copied over my configs, maybe something broke there, we'll see.

**Reply by vladmandic** [reply]
this is code, i don't think config makes any difference. just cant figure out why do we have different results.
and yes, the fact that your exif ends up with json string is the key problem.
oh, just had an idea, will try one more thing.

**Comment by Tillerz** [comment]
Btw. when setting up fresh, it throws a number of errors... same error for everal things:

Comes for LDSR, SwinIR, ScuNet, Lora:

```
13:33:12-219132 ERROR    Module load: X:\git\git_repos\vlad1111\automatic\extensions-builtin\LDSR\preload.py:
                         AssertionError
┌───────────────────────────────────────── Traceback (most recent call last) ──────────────────────────────────
│ X:\git\git_repos\vlad1111\automatic\modules\script_loading.py:13 in load_module
│
│   12 │   try:
│ > 13 │   │   module_spec.loader.exec_module(module)
│   14 │   except Exception as e:
│ in exec_module:883
│
│                                               ... 1 frames hidden ...
│
│ X:\git\git_repos\vlad1111\automatic\extensions-builtin\LDSR\preload.py:2 in <module>
│
│   1 import os
│ > 2 from modules import paths
│   3
│
│ X:\git\git_repos\vlad1111\automatic\modules\paths.py:27 in <module>
│
│    26
│ >  27 assert sd_path is not None, f"Couldn't find Stable Diffusion in any of: {possible_sd_pat
│    28
└──────────────────────────────────────────────────────────────────────────────────────────────────────────────
AssertionError: Couldn't find Stable Diffusion in any of:
['X:\\git\\git_repos\\vlad1111\\automatic\\repositories/stable-diffusion-stability-ai', '.',
'X:\\git\\git_repos\\vlad1111']
```

And then

```
13:36:38-039781 ERROR    Script initialize:
                         X:\git\git_repos\vlad1111\automatic\extensions-builtin\sd-webui-controlnet\scripts\controlnet.
                         py 'Namespace' object has no attribute 'controlnet_preprocessor_cache_size'
13:36:38-102359 ERROR    Script initialize:
                         X:\git\git_repos\vlad1111\automatic\extensions-builtin\sd-webui-controlnet\scripts\controlnet.
                         py 'Namespace' object has no attribute 'controlnet_preprocessor_cache_size'
```

And the last one is:

```
13:37:03-087474 ERROR    executing callback:
                         X:\git\git_repos\vlad1111\automatic\extensions-builtin\sd-webui-controlnet\scripts\api.py
                         app_started_callback: AttributeError
┌───────────────────────────────────────── Traceback (most recent call last) ─────────────────────────────────────
│ X:\git\git_repos\vlad1111\automatic\modules\script_callbacks.py:134 in app_started_callback
│
│   133 │   │   │   t0 = time.time()
│ > 134 │   │   │   c.callback(demo, app)
│   135 │   │   │   timer(t0, c.script, 'app_started')
│
│ X:\git\git_repos\vlad1111\automatic\extensions-builtin\sd-webui-controlnet\scripts\api.py:56 in controlnet_api
│
│   55 │
│ > 56 │   cached_cn_preprocessors = global_state.cache_preprocessors(global_state.cn_preproces
│   57 │   @app.post("/controlnet/detect")
│
│ X:\git\git_repos\vlad1111\automatic\extensions-builtin\sd-webui-controlnet\scripts\global_state.py:23 in
│ cache_preprocessors
│
│    22 │    cache for each preprocessor function. """
│ >  23 │   CACHE_SIZE = shared.cmd_opts.controlnet_preprocessor_cache_size
│    24
└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────
AttributeError: 'Namespace' object has no attribute 'controlnet_preprocessor_cache_size'
```

**Comment by Tillerz** [comment]
Ok. Result. Fresh Setup. Only thing I copied over was the OUTPUT folder and the MODELS.

- Process Image: drop in image
- Click -> Text

Result: The WHOLE JSON BLOCK is being copied into "prompt", including the Negative prompt and all the parameters like seed etc.

So it's not my config or libs.

All I did was clone a1111, copy output and models from old folder and then start it.

So it's broken differently there. Maybe I need a restart after the errors it threw when it started up. Let me do that....

Nah. After restart it updated `Package wrong version: timm 0.4.12 required 0.6.13`, and threw no errors, but same result. Whole JSON + the data after are in PROMPT field. It didn't touch Negative prompt or set any of the other values.

**BUT: with the OLD images that do not use the JSON in image header, it works now.**


If I can help with additional info, just tell me what to do. :)

**Comment by vladmandic** [comment]
> with the OLD images that do not use the JSON in image header, it works now.

JSON should never end up in image headers - that is the core issue. JSON is used internally and it should be translated into infotext when saving the image.

but the thing is, I still cant reproduce.

what are your image options. i'm guessing its something specific to how image is saved to start with.

**Reply by Tillerz** [reply]
Default. I didn't change anything after the reinstall. I started with an empty folder.

**Reply by vladmandic** [reply]
i'll run a fresh install and try to reproduce.

**Comment by vladmandic** [comment]
and lets continue in #1356, discussion thread is not good for this as once its marked as read it totally falls of my radar and i cannot track what's resolved and whats not.
